### PR TITLE
[excel] (Range) Dynamic array sample

### PR DESCRIPTION
**Pull request by AlexJerabek**
Here's the gist link so you can try it out in Script Lab: https://gist.github.com/AlexJerabek/8889dee05873b05f85175e98ff38fd63

This sample shows how to use the new APIs around Range spilling to handle dynamic array content.

Note that this still needs the updates to the Excel mapping workbook for reference snippets. It is waiting on #472 to be checked in to avoid merge conflicts.

### CODE REVIEW

File: samples/excel/85-preview-apis/dynamic-arrays.yaml [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+          } catch (error) {
+            // Note: In a production add-in, you'd want to notify the user through your add-in's UI.
+            console.error(error);
+          }
+        }
+    language: typescript
+template:
+    content: |-
+        <section class="ms-font-m">
+          <p>This sample shows how to apply formulas that return dynamic arrays and show how to get the relevant information
```
Comment (alison-mk) [inline samples/excel/85-preview-apis/dynamic-arrays.yaml:133]:
```suggestion
          <p>This sample shows how to apply formulas that return dynamic arrays and how to get the relevant information
```

Comment (alison-mk) [reply samples/excel/85-preview-apis/dynamic-arrays.yaml:133]: Intro sentence is unclear to me, is the second 'show' necessary? If so, what is doing the showing?

File: samples/excel/85-preview-apis/dynamic-arrays.yaml [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+          <h3>Try it out</h3>
+          <button id="copy-table-headers" class="ms-Button">
+                  <span class="ms-Button-label">Copy table headers</span>
+              </button>
+          <p />
+          <button id="apply-filter-function" class="ms-Button">
+                <span class="ms-Button-label">Apply FILTER formula</span>
+            </button>
+          <p />
+          <p>Select a cell and press the following button to see any spill information for that cell related to dynamic
```
Comment (alison-mk) [inline samples/excel/85-preview-apis/dynamic-arrays.yaml:154]: I think we should clarify that the user needs to select a new, spilled cell for this to work. Maybe 'Select a spilled cell'?

Comment (alison-mk) [inline samples/excel/85-preview-apis/dynamic-arrays.yaml:154]:
```suggestion
          <p>Select a spilled cell and press the following button to see information for that cell related to dynamic
```

File: samples/excel/85-preview-apis/dynamic-arrays.yaml [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+            arrays.</p>
+          <button id="display-spill-information" class="ms-Button">
+              <span class="ms-Button-label">Display spill information for selected range</span>
```
Comment (alison-mk) [inline samples/excel/85-preview-apis/dynamic-arrays.yaml:157]: Since this button only works when a single cell is selected, should the button say 'selected range'? I'm still learning Excel terms so not sure! Along the same lines, should the console readouts say 'selected range' when we're only referring to a single cell?

File: samples/excel/85-preview-apis/dynamic-arrays.yaml (hunk 1 above)
Comment (AlexJerabek) [reply samples/excel/85-preview-apis/dynamic-arrays.yaml:133]: Nope! Good catch.

File: samples/excel/85-preview-apis/dynamic-arrays.yaml (hunk 3 above)
Comment (AlexJerabek) [reply samples/excel/85-preview-apis/dynamic-arrays.yaml:157]: I'll rephrase this to be more clear. It is a constant source of tension in the writing that a range can be a single cell.
